advection_fv4/interface.py

```diff
@@ -0,0 +1,302 @@
+import numpy as np
+from numba import njit
+
+
+@njit(cache=True)
+def states(a, qx, qy, ng, idir):
+    r"""
+    Predict the cell-centered state to the edges in one-dimension using the
+    reconstructed, limited slopes. We use a fourth-order Godunov method.
+
+    Our convention here is that:
+
+        ``al[i,j]``   will be :math:`al_{i-1/2,j}`,
+
+        ``al[i+1,j]`` will be :math:`al_{i+1/2,j}`.
+
+    Parameters
+    ----------
+    a : ndarray
+        The cell-centered state.
+    qx, qy : int
+        The dimensions of `a`.
+    ng : int
+        The number of ghost cells
+    idir : int
+        Are we predicting to the edges in the x-direction (1) or y-direction (2)?
+
+    Returns
+    -------
+    out : ndarray, ndarray
+        The state predicted to the left and right edges.
+    """
+
+    al = np.zeros((qx, qy))
+    ar = np.zeros((qx, qy))
+
+    a_int = np.zeros((qx, qy))
+    dafm = np.zeros((qx, qy))
+    dafp = np.zeros((qx, qy))
+    d2af = np.zeros((qx, qy))
+    d2ac = np.zeros((qx, qy))
+    d3a = np.zeros((qx, qy))
+
+    C2 = 1.25
+    C3 = 0.1
+
+    nx = qx - 2 * ng
+    ny = qy - 2 * ng
+    ilo = ng
+    ihi = ng + nx
+    jlo = ng
+    jhi = ng + ny
```

Comment:
can we get away with just passing in the grid object now?  then we would not need to recompute these

Reply:
We could - I mainly left them in so that the python and fortran functions looked as similar as possible and had the same inputs

Reply:
okay.  Perhaps then after the merge and once we eliminate the Fortran stuff, we can simplify things then.